Apps/Immunization/test/unit/Services.Test/VaccineStatusServiceTests.cs

```diff
@@ -99,15 +110,295 @@ public void ShouldGetVaccineStatus()
+        /// <param name="state">final state.</param>
+        [Theory]
+        [InlineData("Exempt", VaccineState.Exempt)]
+        [InlineData("PartialDosesReceived", VaccineState.PartialDosesReceived)]
+        [InlineData("AllDosesReceived", VaccineState.AllDosesReceived)]
+        public void ShouldGetAuthenticatedVaccineStatus(string statusIndicator, VaccineState state)
+        {
+            RequestResult<PHSAResult<VaccineStatusResult>> delegateResult = new()
+            {
+                ResultStatus = Common.Constants.ResultType.Success,
+                ResourcePayload = new PHSAResult<VaccineStatusResult>()
+                {
+                    LoadState = new PHSALoadState() { RefreshInProgress = false, BackOffMilliseconds = 500 },
+                    Result = new VaccineStatusResult()
+                    {
+                        FirstName = "Bob",
+                        LastName = "Test",
+                        Birthdate = this.dob,
+                        StatusIndicator = statusIndicator,
+                        FederalVaccineProof = new(),
+                    },
+                },
+            };
+            JWTModel jwtModel = new()
+            {
+                AccessToken = this.accessToken,
+            };
+
+            RequestResult<VaccineStatus> expectedResult = new RequestResult<VaccineStatus>()
+            {
+                ResultStatus = delegateResult.ResultStatus,
+                ResourcePayload = new VaccineStatus()
+                {
+                    Loaded = true,
+                    RetryIn = 0,
+                    FirstName = "Bob",
+                    LastName = "Test",
+                    Birthdate = this.dob,
+                    State = state,
+                    FederalVaccineProof = new(),
+                },
+            };
+
+            Mock<IVaccineStatusDelegate> mockDelegate = new Mock<IVaccineStatusDelegate>();
+            mockDelegate.Setup(s => s.GetVaccineStatus(It.IsAny<VaccineStatusQuery>(), this.accessToken, false)).Returns(Task.FromResult(delegateResult));
+
+            Mock<IAuthenticationDelegate> mockAuthDelegate = new Mock<IAuthenticationDelegate>();
+            mockAuthDelegate.Setup(s => s.AuthenticateAsUser(It.IsAny<Uri>(), It.IsAny<ClientCredentialsTokenRequest>())).Returns(jwtModel);
+
+            IVaccineStatusService service = new VaccineStatusService(
+                this.configuration,
+                new Mock<ILogger<VaccineStatusService>>().Object,
+                mockAuthDelegate.Object,
+                mockDelegate.Object,
+                GetMemoryCache(),
+                this.GetHttpContextAccessor().Object);
+
+            var actualResult = Task.Run(async () => await service.GetAuthenticatedVaccineStatus(this.hdid).ConfigureAwait(true)).Result;
             Assert.True(expectedResult.IsDeepEqual(actualResult));
         }
 
+        /// <summary>
+        /// GetVaccineStatus - when the status when there is data mismatch.
```

Comment:
The comment should reference `GetPublicVaccineStatus` and the description should be updated to be clearer.